src/content/docs/Standard Library/index.mdx

```diff
@@ -334,31 +334,116 @@ moving the output pointer 1 or 2 steps.
 ## std::io
 
 ### String! readline(stream = io::stdin(), Allocator allocator = allocator::heap())
-Read from a stream (default is stdin) to the next "\n" or to the end of the stream, whatever comes first.
+Read a `String!` from a file stream. Defaults to standard input (stdin). Reads to the next newline or to the end of stream. `Readline` returns an optional, and must be handled accordingly.
 
+```c
+import std::io;
+
+fn void! main() {
+    String! name; 
+
+    name = io::readline();
+    if (catch excuse = name) {
+        return excuse?;
+    }
+
+    io::printfn("The Matrix has you... Follow the white rabbit. Knock, knock, %s.", name);
+}
+```
+
+Note:
 `\r` will be filtered from the String.
 
 ### String! treadline(stream = io::stdin())
-Same as `readline` but uses the temporary allocator.
+Read a `String!` from a file stream. Defaults to standard input (stdin). Reads to the next newline or to the end of stream. `Treadline` returns an optional, and must be handled accordingly. Additionally, the difference from the regular `readline`, is that this uses the temporary allocator.
 
-### void print(x), void printn(x = "")
-Print any value to stdout.
+```c
+import std::io;
 
+fn void! main() {
+    String! name; 
+
+    name = io::treadline();
+    if (catch excuse = name) {
+        return excuse?;
+    }
+
+    io::printfn("Hello %s! Hope you have a great day", name);
+}
+```
+
+Note:
+`\r` will be filtered from the String.
+
+### void print(x), void printn(x = "")
+Print a value to stdout. Works for different types. 
 The `printn` variant appends a newline.
+```c
+import std::io;
+
+enum Heat {
+    WARM,
+    WARMER,
+    REALLY_WARM,
+}
+
+fn void main() {
+    int[<2>] vec = {4, 2};
+    Heat weather = WARM;
+    int[5] fib = { 0, 1, 1, 2, 3 };
+    String dialogue = "I will never tell you about the muffin man";
+
+    io::print("Hello");   /* Outputs: Hello                                      */
+    io::print(20);        /* Outputs: 20                                         */
+    io::print(2.2);       /* Outputs: 2.200000                                   */
+    io::print(vec);       /* Outputs: [<4, 2>]                                   */
+    io::print(weather);   /* Outputs: WARM                                       */
+    io::print(fib);       /* Outputs: [0, 1, 1, 2, 3]                            */
+    io::print(dialogue);  /* Outputs: I will never tell you about the muffin man */
+}
+```
```

Comment:
```suggestion
### void print(x), void printn(x = "")

Print a value to stdout works for the majority of types, including structs, which can be helpful for debugging.
The `printn` variant appends a newline.

```c3
import std::io;

enum Heat 
{
    WARM,
    WARMER,
    REALLY_WARM,
}

fn void main() 
{
    int[<2>] vec = { 4, 2 };
    Heat weather = WARM;
    int[5] fib = { 0, 1, 1, 2, 3 };
    String dialogue = "secret";

    io::print("Hello");   // Hello
    io::print(20);        // 20
    io::print(2.2);       // 2.200000
    io::print(vec);       // [<4, 2>]
    io::print(weather);   // WARM
    io::print(fib);       // [0, 1, 1, 2, 3] 
    io::print(dialogue);  // secret
}
```
```
docs style is `{` on newline, Line length optimised, using `//` style comments